### PR TITLE
Added tests of Migrator logging messages.

### PR DESCRIPTION
**Pull request by emystein**
Tests that verify SLF4J logging events triggered from Migrator class. Tests use the [slf4j-test](http://projects.lidalia.org.uk/slf4j-test/) library.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -47,6 +48,12 @@
                 </exclusion>
             </exclusions>
         </dependency>
+		<dependency>
```
Comment (nbozic) [inline pom.xml]: Can you use spaces instead of tabs?

File: src/test/java/io/smartcat/migration/MigratorLoggingTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		String expectedMessage = "Migration [{}] to version {} finished in {} seconds.";
+		return info(expectedMessage, migration.getDescription(), migration.getVersion());
+	}
+
+	private LoggingEvent skippedMigrationLoggingEvent(Migration migration) {
+		String expectedMessage = "Skipping migration [{}] with version {} since db is on higher version {}.";
+		int currentVersion = versioner.getCurrentVersion(migration.getType());
+		return warn(expectedMessage, migration.getDescription(), migration.getVersion(), currentVersion);
+	}
+
```
Comment (nbozic) [inline src/test/java/io/smartcat/migration/MigratorLoggingTest.java]: Same goes here for spaces instead of tabs.

**Comment by nbozic** [comment]
Thanks for contribution again!

Can you just change code to use spaces instead of tabs? 

And more general, what triggered you to test order and way of logging? Can you maybe elaborate that in description of pull request.

**Comment by emystein** [comment]
@nbozic, thanks for the review.

Regarding the code format, yes I can change the Eclipse formatter for using spaces instead of tabs. Would you share with me your Eclipse formatter settings so I can apply exactly the same format when I work on this code?

Regarding the logging tests, I wanted to add regresion tests for logic but the easiest way I found for verifying the state changes in the logic was testing the log events, since I had some trouble trying to reproduce Session calls using mocks. Later on I had more sucess mocking the Session so I could discard this commit and change the tests to use a mocked Session.

**Comment by nbozic** [comment]
I will wrap it up since I am in between project changes so I do not remember what we have used for this one, spaces instead of tabs is the only thing I saw (think we used pretty much eclipse standard one with that change). I will send you when I have something.

Regarding mocks on session, on our showcase project, [Spring Cassandra Showcase](https://github.com/smartcat-labs/spring-cassandra-showcase-application) we have stubbed [Session](https://github.com/smartcat-labs/spring-cassandra-showcase-application/tree/master/api/src/test/java/io/smartcat/spring/cassandra/showcase/test/cassandra/stub), and changed it for test purpose to issue sync statements instead of async so we can test executeAsync as well. Alternative for executeAsync os to put thread sleep or something like that which is not nice and for test purpose you just need to verify if data is in DB or not. Maybe you can check that out to get ideas and stub session to store some kind of log messages for verification (this way we do not need to start embedded Cassandra and have slow tests). What do you think?

**Comment by emystein** [comment]
I aggree on stubbing Cassandra for making tests run faster, altough I think we should still have tests that runs against a concrete Cassandra instance, at least an embedded one.
Regarding these changes, please hold on to merge them, maybe I can refactor them to use stubs instead of log message verification. I'd get rid of the logging tests if I succeed with the stubs/mocks.

**Pull request closed without merging**